src/viewers/VivViewer.js

```diff
@@ -119,12 +144,17 @@ export default class VivViewer extends PureComponent {
       return null;
     }
     let dataCoords;
-    // This is currently a work-around for: https://github.com/visgl/deck.gl/pull/4526.
-    // Once this is fixed we can make something more robust.
-    if (sourceLayer.id.includes('Background')) {
-      const { numLevels } = layer.props.loader;
+    // Tiled layer needs a custom layerZoomScale.
+    if (sourceLayer.id.includes('Tiled')) {
+      const {
+        loader: { tileSize }
+      } = layer.props;
+      const { z } = sourceLayer.props;
       // The zoomed out layer needs to use the fixed zoom at which it is rendered (i.e numLevels - 1).
-      const layerZoomScale = Math.max(1, 2 ** Math.floor(numLevels - 1));
+      const layerZoomScale = Math.max(
+        1,
+        2 ** Math.round(-z + Math.log2(512 / tileSize))
+      );
```

Comment:
This fixes a problem that we were having with picking the right number under the mouse.